src/main/scala/io/circe/yaml/parser/NodeAlg.scala

```diff
@@ -0,0 +1,160 @@
+  }
+  
+  def fromFields(ts: Iterable[(String, Either[ParsingFailure, A])]): Either[ParsingFailure, A] = try {
+    Either.right {
+      lifted.fromFields {
+        ts.map {
+          case (key, value) => key -> value.valueOr(throw _)
+        }
+      }
+    }
+  } catch {
+    case f @ ParsingFailure(_, _) => Either.left(f)
+  }
+}
+
+final class AccumlatingAlg[A](base: NodeAlg[A]) extends NodeAlg[ValidatedNel[ParsingFailure, A]] {
+  private val lifted = new LiftedAlg(base)
+  def int(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.int(node).toValidatedNel
+  def float(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.float(node).toValidatedNel
+  def timestamp(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.timestamp(node).toValidatedNel
+  def bool(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.bool(node).toValidatedNel
+  def yNull(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.yNull(node).toValidatedNel
+  def string(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.string(node).toValidatedNel
+  def otherScalar(node: ScalarNode): ValidatedNel[ParsingFailure, A] = lifted.otherScalar(node).toValidatedNel
+
+  def fromFields(ts: Iterable[(String, ValidatedNel[ParsingFailure, A])]): ValidatedNel[ParsingFailure, A] =
+    ts.toList.traverseU {
+      case (key, value) => value.map(key -> _)
+    }.map(base.fromFields)
+
+  def fromValues(ts: Iterable[ValidatedNel[ParsingFailure, A]]): ValidatedNel[ParsingFailure, A] =
+    ts.toList.sequenceU.map(base.fromValues)
+}
+
+class DefaultAlg extends NodeAlg[Json] {
+  protected object Constructor extends SafeConstructor {
+    def flatten(node: MappingNode): Unit = flattenMapping(node)
+  }
+
+  final protected def number(str: String): Json = JsonNumber.fromString(str).map(Json.fromJsonNumber).getOrElse {
+    throw new NumberFormatException(s"Invalid numeric string $str")
+  }
+
+  def int(node: ScalarNode): Json = number(node.getValue)
+  def float(node: ScalarNode): Json = number(node.getValue)
+  def timestamp(node: ScalarNode): Json = Json.fromString(node.getValue)
+  def bool(node: ScalarNode): Json = Json.fromBoolean(node.getValue.toBoolean)
+  def yNull(node: ScalarNode): Json = Json.Null
+  def string(node: ScalarNode): Json = Json.fromString(node.getValue)
+  def otherScalar(node: ScalarNode): Json = if (!node.getTag.startsWith(Tag.PREFIX)) {
+    Json.fromJsonObject(JsonObject.singleton(node.getTag.getValue.stripPrefix("!"), Json.fromString(node.getValue)))
+  } else Json.fromString(node.getValue)
+
+  def fromValues(ts: Iterable[Json]): Json = Json.fromValues(ts)
+  def fromFields(ts: Iterable[(String, Json)]): Json = Json.fromFields(ts)
+}
+
+case class ConfiguredAlg(
+  numericTimestamps: Boolean
+) extends DefaultAlg {
+  final override def timestamp(node: ScalarNode): Json = if (!numericTimestamps) {
+    super.timestamp(node)
+  } else {
+    val constructor = new SafeConstructor.ConstructYamlTimestamp()
```

Comment:
Unfortunately they don't expose just the timestamp parsing logic - otherwise we could bypass this allocation entirely.

Since the spec is pretty fixed, maybe it's worthwhile just duplicating their regexes and doing it internally here?